src/Dom/Element.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace AmpProject\Dom;
+
+use AmpProject\Attribute;
+use AmpProject\Exception\MaxCssByteCountExceeded;
+use AmpProject\Optimizer\CssRule;
+use DOMElement;
+
+/**
+ * Class AmpProject\Dom\Element.
+ *
+ * @property Document $ownerDocument        The ownerDocument for these elements should always be a Dom\Document.
+ * @property int      $inlineStyleByteCount Number of bytes that are consumed by the inline style attribute.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class Element extends DOMElement
+{
+
+    /**
+     * Error message to use when the __get() is triggered for an unknown property.
+     *
+     * @var string
+     */
+    const PROPERTY_GETTER_ERROR_MESSAGE = 'Undefined property: AmpProject\\Dom\\Element::';
+
+    /**
+     * Add CSS styles to the element as an inline style attribute.
+     *
+     * @param string $style CSS style(s) to add to the inline style attribute.
+     * @throws MaxCssByteCountExceeded If the allowed max byte count is exceeded.
+     */
+    public function addInlineStyle($style)
```

Comment:
What if a sanitizer does `setAttribute('style', '...')`? For example: 

https://github.com/ampproject/amp-wp/blob/06c9b9531874174facfa805cad683c73617360a8/includes/sanitizers/class-amp-base-sanitizer.php#L723

Should this not be implemented to route to `addInlineStyle()` (with a recursion check of course)? 

Otherwise, it would be important that `addInlineStyle` never be called until the last `style` attribute is set directly, otherwise the `inlineStyleByteCount` will not be correct.

Another complication: if someone does `$attrNode = $element->getAttributeNode('style')` and then sets the value via `$attrNode->nodeValue`. That would be another way for the byte `inlineStyleByteCount` to become out of sync.

Reply:
Additionally, what if someone creates a `<style>` element and appends it to the document? The style sanitizer will process those in the AMP plugin, but once passed over to the Optimizer then custom transformers could do that. Though I guess they could also add arbitrary other invalid elements, as well as add invalid `style` attributes. So it would be the responsibility of transformers to not do bad things?

Reply:
Regarding the use of `setAttribute('style', ...)`, I think it is irrelevant if it happens before the style sanitizer ran. So, up until that point, it is probably just wasted processing time to calculate any byte lengths.

However, what I could do is to route everything through `addInlineStyle()`, but then have a flag I can set to start enforcing at a certain point. So the style sanitizer would basically say `$document->enforce_css_max_byte_count( 75000 )` and from this point onwards, all changes would be checked for size.

The question here is mainly whether we want to enforce the byte count across our own code, or across any future third-party extensions. Given this is a library for general consumption, it seems to make sense to go that extra mile.

Reply:
> So the style sanitizer would basically say `$document->enforce_css_max_byte_count( 75000 )` and from this point onwards, all changes would be checked for size.

Yes, I like that idea.

Reply:
I've enforced `Dom\Element::setAttribute()` now to always keep the byte count in check once the document has an enforced limit.

I didn't add any logic to the style attribute itself (the `DOMAttr`). I'm a bit wary of extended that deep in the hierarchy for performance reasons. Do you consider `setAttribute()` to be safe enough, or should we go all in? I'd suggest staying away from extending `DOMAttr` for now and see how the extension to `DOMElement` plays out first. The entire PHP DOM subsystem always seems to be up for a few surprises...

Reply:
Yes, I think this is fine for now. It would be unusual for someone to be manipulating a `DOMAttr` directly.